galley/pkg/config/event/selector.go

```diff
@@ -15,41 +15,123 @@
 package event
 
 import (
+	"fmt"
+
 	"istio.io/istio/galley/pkg/config/collection"
 	"istio.io/istio/galley/pkg/config/scope"
 )
 
-// Selector distributes events to multiple different handlers, based on collection name.
-type Selector struct {
-	handlers map[collection.Name]Handler
+// Selector distributes events to different handlers, based on collection name.
+type Selector interface {
+	Handler
+	Broadcast(e Event)
```

Comment:
AFAICT the only type of event that is broadcast is a RESET. Could we just make a selector handle resets properly and get rid of `Broadcast` altogether?  Then we don't even need the `Broadcast` interface and can just define a specific type of `Handler` called a `Router`.

Reply:
Can't the nodes that need to understand FullSync and Reset just do so?

Reply:
The scenario is this: A node has 2 inputs (`a`, `b`) and 3 outputs (`z`,`y`,`x`). How do you handle FullSync? Is FullSync of `a` is enough to publish? 
The node can be only in FullSync mode if both `a` and `b` is in FullSync mode. It will need to signal to the downstream collections that it is in FullSync, so it will need to broadcast that to `z`, `y`, `x`. Reset is a simpler version of this.